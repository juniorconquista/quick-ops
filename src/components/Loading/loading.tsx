import React, { memo } from 'react';
import { layout, space, variant, VariantArgs } from 'styled-system';
import styled, { keyframes } from 'styled-components';

import { LoadingStyledProps } from './loading.types';

const loading = keyframes`
  0% {
    top: 36px;
    left: 36px;
    width: 0;
    height: 0;
    opacity: 1;
  }
  100% {
    top: 0px;
    left: 0px;
    width: 72px;
    height: 72px;
    opacity: 0;
  }
`;

const variants = variant({
    prop: 'styling',
    variants: {
        base: {
            '.loading-screen': {
                position: 'absolute',
                display: 'flex',
                justifyContent: 'center',
                alignItems: 'center',
                top: '0',
                bottom: '0',
                left: '0',
                right: '0',
                zIndex: 99999,
                bg: (props: { colors: { white: { default: string } } }) =>
                    `${props.colors.white.default}80`,
                '.loading': {
                    display: 'inline-block',
                    position: 'relative',
                    width: '80px',
                    height: '80px',
                    '& div': {
                        position: 'absolute',
                        border: '4px solid',
                        borderColor: 'text.default',
                        opacity: '1',
                        borderRadius: '50%',
                    },
                    '& div:nth-child(2)': {
                        animationDelay: '-0.5s',
                    },
                },
            },
        },
    },
} as VariantArgs);

const Style: React.FC<LoadingStyledProps> = styled.div<LoadingStyledProps>`
    ${variants}
    ${space}
    ${layout}
    .loading div {
        animation: ${loading} 1s cubic-bezier(0, 0.2, 0.8, 1) infinite;
    }
`;

const Loading: React.FC = () => (
    <Style styling="base">
        <div className="loading-screen">
            <div className="loading">
                <div></div>
                <div></div>
            </div>
        </div>
    </Style>
);

export default memo(Loading);
