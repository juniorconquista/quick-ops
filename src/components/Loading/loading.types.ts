import { LayoutProps } from 'styled-system';

export type LoadingStyledProps = LayoutProps & {
    styling?: 'base';
};

export interface LoadingProps {
    isOpen: boolean;
}
