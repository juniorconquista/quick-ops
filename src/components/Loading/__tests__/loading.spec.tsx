import React from 'react';
import { cleanup, render } from '@testing-library/react';

import ThemeProvider from '../../../utils/ThemeProviderTest';
import Loading from '../loading';
import 'jest-styled-components';

describe('Loading component', () => {
    afterEach(() => {
        jest.clearAllMocks();
        cleanup();
    });

    it('tests basic snapshot', () => {
        const container = render(
            <ThemeProvider>
                <Loading />
            </ThemeProvider>,
        );
        expect(container.asFragment()).toMatchSnapshot();
    });
});
