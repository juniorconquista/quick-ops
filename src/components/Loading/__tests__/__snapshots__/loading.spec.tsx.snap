// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component tests basic snapshot 1`] = `
<DocumentFragment>
  .c0 .loading-screen {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 99999;
  background-color: #25262a80;
}

.c0 .loading-screen .loading {
  display: inline-block;
  position: relative;
  width: 80px;
  height: 80px;
}

.c0 .loading-screen .loading div {
  position: absolute;
  border: 4px solid;
  border-color: #f8f9fa;
  opacity: 1;
  border-radius: 50%;
}

.c0 .loading-screen .loading div:nth-child(2) {
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

.c0 .loading div {
  -webkit-animation: dnpXZJ 1s cubic-bezier(0,0.2,0.8,1) infinite;
  animation: dnpXZJ 1s cubic-bezier(0,0.2,0.8,1) infinite;
}

<div
    class="c0"
  >
    <div
      class="loading-screen"
    >
      <div
        class="loading"
      >
        <div />
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
